wagtail/core/fields.py

```diff
@@ -153,12 +167,29 @@ def get_prep_value(self, value):
             # for reverse migrations that convert StreamField data back into plain text
             # fields.)
             return value.raw_text
-        else:
+        elif isinstance(value, StreamValue) or not self.use_json_field:
+            # StreamValue instances must be prepared first.
+            # Before use_json_field was implemented, this is also the value used in queries.
```

Comment:
Without checking `isinstance(value, StreamValue)`, queries with `JSONField` features e.g.
```py
Page.objects.filter(body__contains=[{"type": "paragraph_block"}])
```
would fail because `stream_block.get_prep_value(value)` would call `value.get_prep_value()`, where value is `[{"type": "paragraph_block"}]`.

https://github.com/wagtail/wagtail/blob/d149a27392f85efdbee1a28c5988c2f3d6393fc7/wagtail/core/blocks/stream_block.py#L277-L286